dashboard/test/controllers/api/v1/census/census_controller_test.rb

```diff
@@ -49,6 +50,45 @@ class Api::V1::Census::CensusControllerTest < ActionController::TestCase
     refute response['census_submission_id'].nil?, "census_submission_id expected in response: #{response}"
   end
 
+  test 'HoC census submission uses current_census_year' do
+    post :create,
+      params: {
+        form_version: 'CensusHoc2017v3',
+        nces_school_s: '60000113717',
+        submitter_email_address: "fake@email.address",
+        submitter_name: "Somebody",
+        opt_in: true
+      }
+    assert_response 201
+    response = JSON.parse(@response.body)
+    submission_id = response['census_submission_id']
+    submission = Census::CensusSubmission.find(submission_id)
+    refute submission.nil?
+    assert_equal current_census_year, submission.school_year
+  end
+
+  test 'yourschool census submission uses current_census_year' do
+    post :create,
+      params: {
+        form_version: 'CensusYourSchool2017v4',
+        nces_school_s: '60000113717',
+        submitter_email_address: "fake@email.address",
+        submitter_name: "Somebody",
+        submitter_role: 'OTHER',
+        how_many_do_hoc: 'NONE',
+        how_many_after_school: 'NONE',
+        how_many_10_hours: 'NONE',
+        how_many_20_hours: 'NONE',
+        opt_in: true
+      }
+    assert_response 201, @response.body.to_s
+    response = JSON.parse(@response.body)
+    submission_id = response['census_submission_id']
+    submission = Census::CensusSubmission.find(submission_id)
+    refute submission.nil?
+    assert_equal current_census_year, submission.school_year
```

Comment:
Ah, clever! I'm wondering, though, if we should use Timecop to document intended behavior of current_census_year (so timecop a day earlier/later in the year and hard-code the year it should be) –– that logic is not tested anywhere. But I could also see an argument not to do that since we don't have confirmation of the behavior we want for the year ... we could do that as part of the follow-up work item when we confirm desired behavior of school year?

Reply:
yeah, let's add that test coverage once we confirm whether current_census_year is the behavior we want.

Reply:
apologies -- we SHOULD use timecop here, to make sure this test doesn't break in July/August (which it looks like it would). but I don't think we need to add detailed testing of current_census_year here.

Reply:
Ah yes, good call re: ensuring no flakey tests!